Classes/Util.php

```diff
@@ -462,11 +462,24 @@ public static function initializeTsfe(
             // for certain situations we need to trick TSFE into granting us
             // access to the page in any case to make getPageAndRootline() work
             // see http://forge.typo3.org/issues/42122
-            $pageRecord = BackendUtility::getRecord('pages', $pageId, 'fe_group');
+            $pageRecord = BackendUtility::getRecord('pages', $pageId, 'fe_group, pid');
+
+            $GLOBALS['TSFE']->sys_page = GeneralUtility::makeInstance(PageRepository::class);
```

Comment:
Thanks for your patch and the feedback.

I have two comments:

* i think i should be possible to instanciate the PageRepository only inside the if, since it was not used before and is only needed in that case
* can you add an integration tests that cores this part (that fails before an i green afterwards, since this is a very sensible part and more an edge use case and we want to make sure to not break other stuff with this change?